### ISSUE TITLE
In Server-side pagination mode, the page number doesn't switch on view.

### ISSUE THREAD

**Issue by b1gdanhe** [HC200ok/vue3-easy-data-table#232]
When I request the backend, the responses in ok and i update the servers options.

All things seen work good, but at next page the number of page doesn't change and at the end is not possible to come back to first page unless I reload the view

![Screenshot from 2023-02-19 01-22-59](https://user-images.githubusercontent.com/93092200/219905562-680eafb6-eadf-4739-87bb-b7d92c5cab6f.png)
![Screenshot from 2023-02-19 01-23-15](https://user-images.githubusercontent.com/93092200/219905565-68227b9a-196a-48fb-8806-409b51430373.png)
![Screenshot from 2023-02-19 01-23-27](https://user-images.githubusercontent.com/93092200/219905566-d7534154-6488-4c36-b6c8-05287eef48ba.png)

**Comment by HC200ok** [comment]
Could you provide more details? It will be better to have an online reproduction on `codesandbox` and stuff.

**Comment by b1gdanhe** [comment]
Ok, let me try to put it on codesanbox.

**Comment by b1gdanhe** [comment]
This is the links of the sand box [Test](https://b0n0nm-5173.preview.csb.app/)

**Comment by HC200ok** [comment]
Did you use `server-items-length` and `loading` props?

Here is a demo you can refer to: https://codesandbox.io/s/server-side-paginate-and-sort-mif1nr?file=/src/App.vue:117-180

**Comment by b1gdanhe** [comment]
I solve it.

But the problem, is that the EasyDataTable component don't totally update the serverOptions props inside itself So the try to re-render the component after the switching on new page and it worked.

Thank you for your contribution.

**Comment by pzoechner** [comment]
I encounter the same issue. Apart from the mock server, I'm followed the docs. Making use of `server-items-length` and `loading` as well. Using the Laravel paginator server-side. Like @b1gdanhe said, it does seem like the footer portion is just not updating properly.

**Comment by HC200ok** [comment]
@pzoechner If you are rendering`vue3-easy-data-table` on the server side template like `phtml` instead of using restful API on the browser side. I suggest you re-render the component as @b1gdanhe did https://github.com/HC200ok/vue3-easy-data-table/issues/232#issuecomment-1437652612

**Comment by HC200ok** [comment]
@b1gdanhe `vue3-easy-data-table` updates `serverOptions` in javascript (client side). On the server side, you are supposed to re-render the page to update the `vue3-easy-data-table`.

**Comment by pzoechner** [comment]
I'm rendering `vue3-easy-data-table` on the client-side (Inertia + Vue3 to be exact.) Just wanted to make clear that I'm also having this issue using server-side mode same as OP.
The `serverOptions` prop is updating as it should but as described by @b1gdanhe, the pagination information in the footer isn't.

**Comment by HC200ok** [comment]
@pzoechner Could you please provide demo code for reproduction？

**Comment by b1gdanhe** [comment]
@pzoechner in practice, My Vue app send the request to my backend, and when I get the responses, I bind all necessary data and I increment one variable that I defined as key on the `vue3-easy-data-table` component. Something like this :

`<easy-data-table :key="reRender" ></easy-data-table>`

After the request in the end of my watcher, I just do `reRender++`. I force the Vue app to re-Rend the component.

I also found help in this article [](https://michaelnthiessen.com/force-re-render/) about my rereading technique
